aidial_adapter_openai/gpt4_multi_modal/download.py

```diff
@@ -23,30 +24,55 @@
 
 def guess_attachment_type(attachment: dict) -> Optional[str]:
     type = attachment.get("type")
-    if type is None:
-        return None
 
-    if "octet-stream" in type:
+    if type is None or "octet-stream" in type:
```

Comment:
Could you please explain reasoning of such change?

I understand, that there is a possibility of guessing file type correctly for such cases

```json
{
"type": null,
"url": "http://test.com/image.png"
}
```

But I think it's better to enforce user to set types for attachments, as we did before, don't you think?

Reply:
What we did before was too restrictive.

`type` is an optional field, meaning we have to our best to figure out the type from `url` or `data` when it's absent, otherwise it did make any sense to make it optional in the first place.

Consider it's being `None` as the same as it's being `application/octet-stream` - it's declaration of the user's ignorance about the type of the attachment.